src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -77,6 +77,58 @@ public class SslStreamNetworkStreamTest
             listener.Stop();
         }
 
+        [Fact]
+        public async Task SslStream_SendReceiveOverNetworkStream_AuthenticationException()
+        {
+            TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
+
+            using (X509Certificate2 serverCertificate = Configuration.Certificates.GetServerCertificate())
+            using (TcpClient client = new TcpClient())
+            {
+                listener.Start();
+
+                Task clientConnectTask = client.ConnectAsync(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndpoint).Port);
+                Task<TcpClient> listenerAcceptTask = listener.AcceptTcpClientAsync();
+
+                await Task.WhenAll(clientConnectTask, listenerAcceptTask);
+
+                TcpClient server = listenerAcceptTask.Result;
+                using (SslStream clientStream = new SslStream(
+                    client.GetStream(),
+                    false,
+                    new RemoteCertificateValidationCallback(ValidateServerCertificate),
+                    null,
+                    EncryptionPolicy.RequireEncryption))
+                using (SslStream serverStream = new SslStream(
+                    server.GetStream(),
+                    false,
+                    null,
+                    null,
+                    EncryptionPolicy.RequireEncryption))
+                {
+
+                    Task clientAuthenticationTask = clientStream.AuthenticateAsClientAsync(
+                        serverCertificate.GetNameInfo(X509NameType.SimpleName, false),
+                        null,
+                        SslProtocols.Tls11,
+                        false);
+
+                    AuthenticationException e = await Assert.ThrowsAsync<AuthenticationException>(() => serverStream.AuthenticateAsServerAsync(
+                        serverCertificate,
+                        false,
+                        SslProtocols.Tls12,
+                        false));
+
+                    Assert.NotNull(e.InnerException);
+                    Assert.True(e.InnerException.Message.Contains("SSL_ERROR_SSL"));
```

Comment:
Does this pass on Windows and mac?  I'd have expected this to be Linux-specific.

Reply:
I forgot to add the platform specific attribute... It should fail on anything other than Linux...